core/search/index.py

```diff
@@ -228,8 +229,39 @@ def register_neural_search_fields(
             }
         )
 
-    def reindex(self) -> None:
-        helpers.reindex(self.client, self.name, self.name)
+    def reindex(self, fields: List[str]) -> None:
+        CHUNK_SIZE = 1000
+        THREAD_COUNT = 4
```

Comment:
Here I believe that thread count and chunk size are dependent on the hardware where this is being run right? Do you think we should set them as environment variables or try to detect the correct number of threads based on cpu cores?

Reply:
I tried that, but if the thread count is too high then you risk overwhelming opensearch and it throws `429` errors. I chose these values because they're the same ones used by pgsync.

Reply:
Ok, if pgsync uses those values let's keep this as it is right now, but in the future we might want to make this an environment variable for deploying on different hardware.